lib/msf/core/module/external.rb

```diff
@@ -119,8 +119,45 @@ def process_report(m, mod)
       })
 
       invalidate_login(**cred)
+
+    when 'credential_login'
+      handle_credential_login(data, mod)
     else
       print_warning "Skipping unrecognized report type #{m.params['type']}"
     end
   end
 end
+
+#
+# Handles login report that does not necessarily need to include a password
+#
+def handle_credential_login(data, mod)
+  # Required
+  service_data = {
+      address: data['address'],
+      port: data['port'],
+      protocol: data['protocol'],
+      service_name: data['service_name'],
+      module_fullname: data['fullname'],
+      workspace_id: myworkspace_id
+  }
+
+  # Optional
+  credential_data = {
+      origin_type: :service,
+      username: data['username']
+  }.merge(service_data)
+
+  if data.has_key?(:password)
+    print_warning "In pass"
```

Comment:
Is this a debug statement from development?

Reply:
bwhaha - thanks